public/app/features/variables/editor/VariableTextAreaField.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { FormEvent, PropsWithChildren, ReactElement, useCallback } from 'react';
+import { InlineField, TextArea, useStyles } from '@grafana/ui';
+import { GrafanaTheme } from '@grafana/data';
+import { css } from 'emotion';
+
+interface VariableTextAreaFieldProps<T> {
+  name: string;
+  value: string;
+  placeholder: string;
+  onChange: (event: FormEvent<HTMLTextAreaElement>) => void;
+  width: number;
+  tooltip?: string;
+  ariaLabel?: string;
+  required?: boolean;
+  labelWidth?: number;
+  onBlur?: (event: FormEvent<HTMLTextAreaElement>) => void;
+}
+
+export function VariableTextAreaField({
+  name,
+  value,
+  placeholder,
+  tooltip,
+  onChange,
+  onBlur,
+  ariaLabel,
+  required,
+  width,
+  labelWidth,
+}: PropsWithChildren<VariableTextAreaFieldProps<any>>): ReactElement {
+  const styles = useStyles(getStyles);
+  const getLineCount = useCallback((value: any) => {
+    if (value && typeof value === 'string') {
+      return value.split('\n').length;
+    }
+
+    return 1;
+  }, []);
+
+  return (
+    <div className="gf-form">
```

Comment:
Do we have any "new" forms component alternative to this one?

Reply:
Oops, that should be a VerticalGroup I think. Will fix

Reply:
Thank you for catching that!